velox/functions/prestosql/tests/RegexpReplaceTest.cpp

```diff
@@ -113,5 +115,158 @@ TEST_F(RegexpReplaceTest, withReplacement) {
   test::assertEqualVectors(expected, result);
 }
 
+TEST_F(RegexpReplaceTest, lambda) {
+  auto input = makeRowVector({
+      makeFlatVector<std::string>(
+          {"new york", "", "los angeles", "san francisco", "seattle"}),
+      makeFlatVector<std::string>({"[a]", "[b]", "[c]", "[d]", "[e]"}),
+      makeFlatVector<int64_t>({0, 1, 2, 3, 4}),
+  });
+
+  auto result = evaluate(
+      "regexp_replace(c0, '(\\w)(\\w*)', x -> (concat(upper(x[1]), lower(x[2]))))",
+      input);
+
+  VectorPtr expected = makeFlatVector<std::string>(
+      {"New York", "", "Los Angeles", "San Francisco", "Seattle"});
+  test::assertEqualVectors(expected, result);
+
+  // No matches for any input string. The result should be the same as input.
+  result = evaluate(
+      "regexp_replace(c0, '\\d+', x -> (concat(upper(x[1]), lower(x[2]))))",
+      input);
+  test::assertEqualVectors(input->childAt(0), result);
+
+  // No matches for the second group.
+  result = evaluate(
+      "regexp_replace(c0, '(\\w)(\\d*)', x -> (concat(upper(x[1]), lower(x[2]))))",
+      input);
+  expected = makeFlatVector<std::string>(
+      {"NEW YORK", "", "LOS ANGELES", "SAN FRANCISCO", "SEATTLE"});
+  test::assertEqualVectors(expected, result);
+
+  // Regex with no matching groups.
```

Comment:
Why is this "regex with no matching groups"? Do you mean the replacement doesn't reference matching groups instead?

Reply:
The regex is '\\w\\w*'. It doesn't have any capture groups (nothing in parentheses), hence, the array of groups passed to lambda is always empty.